webapp/routes/index.js

```diff
@@ -3,76 +3,107 @@ var router = express.Router();
     res.json(job_args);
-    return; 
+    res.status(400);
+    return;
   }
-  var key_pair = hash_generator.generate(
+  let key_pair = hash_generator.generate(
     config.get("secret_token_length"),
     config.get("secret_token_length"),
     config.get("secret_token_salt")
   );
-  job_args.secret = key_pair.secret;
-  var job = job_queue.create("job", job_args)
-  .ttl(config.get("job_defaults.ttl"))
-  .save( function(err){
-    if( !err ) {
-      res.json({ "job_id": job.id, "access_token": key_pair.key });
+  let job_data = {};
+  job_data.secret = key_pair.secret;
+  if (job_args.externalCallbackURL) {
+    job_data.externalCallbackURL = job_args.externalCallbackURL;
+  }
+  kennel.postTask(req.body, (error, response) => {
+    if (error) {
+      res.status(error.code ? error.code : 500);
+      res.json({error: error});
     } else {
-      res.json({})
+      job_data.status = 'queued';
+      Job.persist(response.taskId, job_data);
+      res.status(202);
+      res.json({jobId: response.taskId, accessToken: key_pair.key});
     }
   });
 });
 
-router.get('/job/:id', function(req, res) {
-  kue.Job.get(req.params.id, function(err, job) {
+router.get('/job/:id', async function(req, res) {
+  let tokenValidated = await Job.checkKey(req.params.id, req.query.token);
+  if (!tokenValidated) {
+    res.status(403).json({"error": "missing or invalid token"});
+    return;
+  }
+  kennel.getTask(req.params.id, function(err, body) {
     if (err) {
       res.status(404).json({"error": `unknown job ${req.params.id}`});
       return;
     }
-    if (job_helper.validateJobToken(job, req.query.token)) {
-      res.json(job_helper.getJobStatus(job, job_queue));
-    } else {
-      res.status(403).json({"error": "missing or invalid token"});
-    }
+    res.json(body);
   });
 });
 
-router.get('/job/:id/download', function(req, res) {
+router.get('/job/:id/download', async function(req, res) {
   var redirect = (req.query.redirect === "true");
-  kue.Job.get(req.params.id, function(err, job) {
-    if (err) {
-      res.status(404).json({"error": `unknown job ${req.params.id}`});
-      return;
-    }
-    if (!job_helper.validateJobToken(job, req.query.token)) {
-      res.status(403).json({"error": "missing or invalid token"});
-      return;
-    }
-    if (job.state() !== "complete") {
-      res.status(202).json({
-        "message": `job status ${job.state()}. try again later.`
-      });
-      return;
-    }
-    var downloadURL = job_helper.getJobDownloadURL(job, res, redirect);
-  });
+  let job;
+  try {
+    job = await Job.getJob(req.params.id);
+  } catch (e) {
+    debug(e);
+    return res.status(404).json({"error": `unknown job ${req.params.id}`});
+  }
+  debug('download job', job);
+  let tokenValidated = await Job.checkKey(req.params.id, req.query.token);
+  if (!tokenValidated) {
+    res.status(403).json({"error": "missing or invalid token"});
+    return;
+  }
+  if (job.status !== "success") {
```

Comment:
last minute API cleanup to mimic the previous implementation. still have a few breadcrumbs laying around I think